corehq/apps/userreports/sql/adapter.py

```diff
@@ -133,23 +133,39 @@ def get_distinct_values(self, column, limit):
 
     def _best_effort_save_rows(self, rows, doc):
         try:
-            self._save_rows(rows, doc)
+            self.save_rows(rows)
         except IntegrityError:
             pass  # can be due to users messing up their tables/data so don't bother logging
         except Exception as e:
             self.handle_exception(doc, e)
 
-    def _save_rows(self, rows, doc):
-        rows = [
+    def bulk_save(self, docs):
+        rows = []
+        for doc in docs:
+            rows.extend(self.get_all_values(doc))
+        self.save_rows(rows)
```

Comment:
How does the error handling work here - or is that still to come?

Reply:
Yes, planning to add one separately. There's usually a `save` and `best_effort_save` corresponding to these save methods that do error handling in this code for some reason.